### DISCUSSION TITLE
Custom type and property metadata during declaration

### DISCUSSION THREAD

**Discussion by gramian** [ArcadeData/arcadedb#1759] [Ideas]
Currently one needs to use `ALTER TYPE` or `ALTER PROPERTY` to set a custom metadata key-value pair, ie:
```
CREATE DOCUMENT TYPE doc;
ALTER TYPE doc CUSTOM coolness = 1;

CREATE PROPERTY doc.num INTEGER (readonly true);
ALTER PROPERTY doc.num CUSTOM oflines = 2;
```
It would be more convenient to directly set metadata in `CREATE x TYPE` or `CREATE PROPERTY`. As for `ALTER`, the syntax introducing the metadata pairs can also be introduced with the `CUSTOM` keyword, like:
```
CREATE DOCUMENT TYPE doc CUSTOM coolness = 10;

CREATE PROPERTY doc.num INTEGER (readonly true) CUSTOM oflines = 1;
```
Furthermore, multiple custom key-value pairs should be passable separated by commas, like for attributes:
```
CREATE VERTEX TYPE vex CUSTOM x = 'width', y = 'height';
```

I will look into the SQL definition if this is not too hard and report in this thread, which can also be moved to an issue.

**Comment by gramian** [comment]
@lvca is this something that is possible now?

**Comment by lvca** [comment]
Yes, with the new parser should be easy to do, let me check

**Comment by lvca** [comment]
Started issue https://github.com/ArcadeData/arcadedb/issues/5409

**Reply by lvca** [reply]
Implemented ;-)
